src/test/java/com/github/tomakehurst/wiremock/client/HttpAdminClientTest.java

```diff
@@ -91,4 +95,17 @@ public void reuseConnections() throws InterruptedException, IOException {
     client.resetAll();
     server.stop();
   }
+
+  @Test
+  public void shouldThrowExceptionWithUrlForStubMappingFromNonWireMockServerPort()
+      throws IOException {
+    var nonWireMockServer = HttpServer.create(new InetSocketAddress(0), 0);
+    nonWireMockServer.start();
+    var serverPort = nonWireMockServer.getAddress().getPort();
+    var client = new HttpAdminClient("localhost", serverPort, ADMIN_TEST_PREFIX);
+    var mapping = post(urlPathMatching("/test")).willReturn(ok()).build();
+    var thrown = assertThrows(ClientError.class, () -> client.addStubMapping(mapping));
+    assertThat(thrown.getMessage()).contains("localhost:" + serverPort);
+    nonWireMockServer.stop(0);
+  }
```

Comment:
What about adjusting the test to check the error message does actually match the expectation, further than just checking that contains `localhost`?

Reply:
I extended assertions for thrown exception. I don't want to compare the whole message, cause part of it depends on non WireMock server and is not controlled by this repo, so checked exception details by regex.

e.g. creating a server by `HttpServer.create(..)`, we get the following message:

`Error parsing response body '<h1>404 Not Found</h1>No context found for request' with status code 404 for http://localhost:xyz/admin-test/__admin/mappings. Error: Unexpected character ('<' (code 60)): expected a valid value (JSON String, Number, Array, Object or token 'null', 'true' or 'false')\n at [Source: (String)\"<h1>404 Not Found</h1>No context found for request\"; line: 1, column: 1]`


but with Spring-Boot server in will be absolutely different one:

`Error parsing response body '{\"code\":\"404.null.55\",\"message\":\"Not Found\"}' with status code 404 for http://localhost:xyz/admin-test/__admin/mappings. Error: Unrecognized field \"code\", not marked as ignorable.
`